<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="settingsGroupBox">
      <property name="title">
       <string>Settings</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="3" column="0" colspan="2">
        <widget class="QLabel" name="thresholdLabel">
         <property name="text">
          <string>Threshold</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1" colspan="3">
        <widget class="QSlider" name="slopeSlider">
         <property name="value">
          <number>5</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2" colspan="2">
        <widget class="QLabel" name="minColorLabel">
         <property name="minimumSize">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QRadioButton" name="useCrominanceRadioButton">
         <property name="text">
          <string>Use Crominance</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QRadioButton" name="enableRadioButton">
         <property name="text">
          <string>Enabled</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="slopeLabel">
         <property name="text">
          <string>Slope</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" colspan="2">
        <widget class="QLabel" name="maxColorLabel">
         <property name="minimumSize">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="3" column="2" colspan="2">
        <widget class="QSlider" name="thresholdSlider">
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
         <property name="value">
          <number>10</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QRadioButton" name="maskOutputRadioButton">
         <property name="text">
          <string>Mask Output</string>
         </property>
         <property name="autoExclusive">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1" rowspan="2">
     <widget class="GLImageWidget" name="glImageWidget">
      <property name="minimumSize">
       <size>
        <width>1000</width>
        <height>570</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>540</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="fileOpenAction"/>
    <addaction name="separator"/>
    <addaction name="fileSaveAction"/>
    <addaction name="fileSaveAsAction"/>
    <addaction name="separator"/>
    <addaction name="exitAction"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="aboutAction"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="fileOpenAction">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="fileSaveAction">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="exitAction">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
  <action name="fileSaveAsAction">
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLImageWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>GLImageWidget.h</header>
   <slots>
    <signal>colorChanged()</signal>
    <slot>enableChromaKey(bool)</slot>
    <slot>setThreshold(int)</slot>
    <slot>setSlope(int)</slot>
    <slot>setCrominance(bool)</slot>
    <slot>setMaskOutput(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>exitAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileOpenAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileSaveAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileSaveAsAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutAction</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutDialogShow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>glImageWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>770</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>glImageWidget</receiver>
   <slot>enableChromaKey(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>768</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glImageWidget</receiver>
   <slot>setThreshold(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>768</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slopeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glImageWidget</receiver>
   <slot>setSlope(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>768</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>useCrominanceRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>glImageWidget</receiver>
   <slot>setCrominance(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>775</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maskOutputRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>glImageWidget</receiver>
   <slot>setMaskOutput(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>775</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>aboutDialogShow()</slot>
  <slot>updateColor()</slot>
 </slots>
</ui>
